exo-go/src/docker_setup/index.go

```diff
@@ -0,0 +1,138 @@
+package dockerSetup
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Originate/exosphere/exo-go/src/app_dependency_helpers"
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/os_helpers"
+	"github.com/Originate/exosphere/exo-go/src/service_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	"github.com/Originate/exosphere/exo-go/src/util"
+	"github.com/pkg/errors"
+)
+
+// DockerSetup renders docker-compose.yml file with service configuration
+type DockerSetup struct {
+	AppConfig     types.AppConfig
+	ServiceConfig types.ServiceConfig
+	ServiceData   types.ServiceData
+	Role          string
+	Logger        *logger.Logger
+	AppDir        string
+}
+
+// NewDockerSetup is DockerSetup's constructor
+func NewDockerSetup(appConfig types.AppConfig, serviceConfig types.ServiceConfig, serviceData types.ServiceData, role string, logger *logger.Logger, appDir string) *DockerSetup {
+	return &DockerSetup{
+		AppConfig:     appConfig,
+		ServiceConfig: serviceConfig,
+		ServiceData:   serviceData,
+		Role:          role,
+		Logger:        logger,
+		AppDir:        appDir,
+	}
+}
```

Comment:
I don't think this function is needed since all you do is assign things. In this case, the caller can just create a `DockerSetup` struct directly.